src/main/java/com/cloudant/client/org/lightcouch/CouchDbClient.java

```diff
@@ -486,20 +487,32 @@ public HttpConnection execute(HttpConnection connection) {
                     case HttpURLConnection.HTTP_PRECON_FAILED: //412
                         ex = new PreconditionFailedException(response);
                         break;
+                    default:
+                        ex = new CouchDbException(response, code);
+                        break;
                 }
                 es = connection.getConnection().getErrorStream();
                 //if there is an error stream try to deserialize into the typed exception
                 if (es != null) {
                     Class<? extends CouchDbException> exceptionClass = ex.getClass();
                     try {
-                        ex = getGson().fromJson(new InputStreamReader(es),
+                        //Register an InstanceCreator that returns the existing exception so we can
+                        //just populate the fields, but not ignore the constructor.
+                        //Uses a new Gson so we don't accidentally recycle an exception.
+                        Gson g = new GsonBuilder().registerTypeAdapter(exceptionClass, new
+                                InstanceCreator<CouchDbException>() {
+                                    @Override
+                                    public CouchDbException createInstance(Type type) {
+                                        return ex;
+                                    }
+                                }).create();
+                        //now populate the exception with the error/reason other info from JSON
+                        g.fromJson(new InputStreamReader(es),

```

Comment:
Yes, we should. I think we can set anything that couldn't be parsed as JSON as the error on the exception. I'll get that added in.